packages/nx/src/migrations/update-16-8-0/escape-dollar-sign-env-variables.ts

```diff
@@ -0,0 +1,63 @@
+import { Tree } from '../../generators/tree';
+import { getProjects } from '../../generators/utils/project-configuration';
+
+/**
+ * This function escapes dollar sign in env variables
+ * It will go through:
+ * - '.env', '.local.env', '.env.local'
+ * - .env.[target-name], .[target-name].env
+ * - .env.[target-name].[configuration-name], .[target-name].[configuration-name].env
+ * - .env.[configuration-name], .[configuration-name].env
+ * at each project root and workspace root
+ * @param tree
+ */
+export default async function escapeDollarSignEnvVariables(tree: Tree) {
+  for (const [_, configuration] of getProjects(tree).entries()) {
+    const envFiles = ['.env', '.local.env', '.env.local'];
+    for (const targetName in configuration.targets) {
+      const task = configuration.targets[targetName];
+      envFiles.push(`.env.${targetName}`, `.${targetName}.env`);
+
+      if (task.configurations) {
+        for (const configurationName in task.configurations) {
+          envFiles.push(
+            `.env.${targetName}.${configurationName}`,
+            `.${targetName}.${configurationName}.env`,
+            `.env.${configurationName}`,
+            `.${configurationName}.env`
+          );
+        }
+      }
+    }
+
+    for (const envFile of envFiles) {
+      parseEnvFile(tree, `${configuration.root}/${envFile}`); // parse at project root
+      parseEnvFile(tree, envFile); // parse at workspace root
```

Comment:
Instead of calling it twice here for each item in the loop, let's just call it once and push the items above.

Reply:
changed